pkg/manager/modelregistry.go

```diff
@@ -79,6 +91,69 @@ func (m *Manager) Capabilities() []*gnmi.ModelData {
 	return outputList
 }
 
-func toModelName(name string, version string) string {
+// extractReadOnlyPaths is a recursive function to extract a list of read only paths from a YGOT schema
+func extractReadOnlyPaths(deviceEntry *yang.Entry, parentState yang.TriState, parentPrefix string, parentPath string) []string {
+	readOnlyPaths := make([]string, 0)
+
+	for _, dirEntry := range deviceEntry.Dir {
+		var itemPath string
+		if parentPath == "" {
+			itemPath = fmt.Sprintf("%s/%s:%s", parentPath, dirEntry.Prefix.Name, dirEntry.Name)
+		} else {
+			itemPath = fmt.Sprintf("%s/%s", parentPath, dirEntry.Name)
+		}
+		//fmt.Println("Processing", itemPath)
+		if dirEntry.IsLeaf() {
+			// No need to recurse
+			if dirEntry.Config == yang.TSFalse || parentState == yang.TSFalse {
+				readOnlyPaths = append(readOnlyPaths, itemPath)
+			}
+		} else if dirEntry.IsContainer() {
+			var name string
```

Comment:
I don't think this is needed, you can use := in the if else clause

Reply:
but then it woun't be available outside the **if** statement